protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1456,109 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
+	// Note: currently not implementing array as we are only implementing this in the operation response context
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isQuotedString(defaultValue)) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+		err := isJSONNumber(strings.ToLower(defaultValue), headerType)
+		if err != nil {
+			return err
+		}
+	case
+		"integer":
+		_, err := strconv.ParseInt(defaultValue, 0, 64)
```

Comment:
My point about parsing the `format` as well is that we could use `strconv.ParseInt(defaultValue, 0, 32)` for 32bit integer specifications. As this logic works at the moment, you could specify `integer, int32` and give a default value that is outside of that range and it would not error.

Reply:
Makes sense... added tests for this use case..